dubbo-kubernetes/src/main/java/org/apache/dubbo/registry/kubernetes/KubernetesServiceDiscovery.java

```diff
@@ -168,23 +167,23 @@ public void doUnregister(ServiceInstance serviceInstance) throws RuntimeExceptio
     @Override
     public Set<String> getServices() {
         return kubernetesClient
-            .services()
-            .inNamespace(namespace)
-            .list()
-            .getItems()
-            .stream()
-            .map(service -> service.getMetadata().getName())
-            .collect(Collectors.toSet());
+                .services()
+                .inNamespace(namespace)
+                .list()
+                .getItems()
+                .stream()
+                .map(service -> service.getMetadata().getName())
+                .collect(Collectors.toSet());
     }
 
     @Override
     public List<ServiceInstance> getInstances(String serviceName) throws NullPointerException {
         Endpoints endpoints =
-            kubernetesClient
-                .endpoints()
-                .inNamespace(namespace)
-                .withName(serviceName)
-                .get();
+                kubernetesClient
+                        .endpoints()
+                        .inNamespace(namespace)
+                        .withName(serviceName)
+                        .get();
```

Comment:
This step seems to still needs to interact with api-server directly on every event. 

Is it possible for us to use the informer cache directly on each event notification? Introducing Informer guarantees that all events will be received even on connection loss but the cache hasn't been fully used.

Reply:
I added SERVICE_INSTANCE MAP, which caches the change endpoints from informer notifications.
In this way, it can reduce the number of times to get the endpoint list from k8s at a time.